### PR TITLE
Add project spaces e2e tests

### LINKED ISSUE #6497: E2E tests for spaces
# Description
The spaces feature set is getting bigger and better, but our e2e test coverage is 0%. That needs to change - especially to keep manual smoke test efforts low during releases.

## User Stories

* > As a developer (reva, ocis or web) I'd like to be certain that my changes for spaces don't break the basic feature set, so that I can merge my PR with confidence.

## Value
- release spaces features with confidence

## Acceptance Criteria
- outline the basic spaces feature set in a feature file
- write playwright tests

## Definition of ready
[ ] everybody needs to understand the value written in the user story
[ ] acceptance criteria has to be defined
[ ] all dependencies of the user story need to be identified
[ ] feature should be seen from an end user perspective
[ ] user story has to be estimated
[ ] story points need to be less then 20

## Definition of done
- Functional requirements
[ ] functionality described in the user story works
[ ] acceptance criteria are fulfilled
- Quality
[ ] codre review happened
[ ] CI is green
[ ] critical code received unit tests by the developer
[ ] automated tests passed (if automated tests are not available, this test needs to be created and passed
- Non-functional requirements
[ ] no sonar cloud issues

### PR DESCRIPTION
**Pull request by fschade**
## Description

First iteration on adding e2e tests for the project spaces, the idea was to keep everything compatible and only write new actions when needed.

if ci and reviewers are happy... ready to be merged.

Please see:
* add project spaces tests, pages and actions
* add dedicated key for the internal object stores (set, get)
* add store and environment for spaces
* get rid of external store usage in favor of environments
* add tests-selectors in components and views
* update shared actions to work inside project spaces
* differentiate between oc10, oCis and global tests via glob
* cleanup some internals like interface over type
* refactor api package actions to always contain a user
* always use join for api calls
* remove versionApi parameter from the api request method to freely choose the endpoint
* add settings endpoint to the api
* add graph endpoints to the api

## Related Issue
- Fixes #6497
- Could help #6557
- Could help #6541

## Motivation and Context
increase e2e test coverage

## How Has This Been Tested?
- writing tests that test tests.... ;)

## Types of changes
- [X] Tests

## Checklist:
- [X] Code changes
- [X] E2E tests added

## Open tasks:
- [ ] see if ci is happy, then good to go

### CODE REVIEW

**Comment by fschade** [comment]
Cc.: @ScharfViktor

File: packages/web-app-files/src/views/spaces/Projects.vue [hunk 1]
```diff
@@ -36,6 +36,7 @@
           <li v-for="space in spaces" :key="space.id" class="oc-mb-m">
             <div
               class="spaces-list-card oc-card oc-card-default oc-rounded"
+              :data-space-id="space.id"
```
Comment (pascalwengerter) [inline packages/web-app-files/src/views/spaces/Projects.vue:39]: @fschade this leads to a failing snapshot test 😓 sorry, mate!

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/24350/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-4-1-02-03-51.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-4-1-02-04-11.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/file-alice-2022-4-1-02-02-15.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/file-brian-2022-4-1-02-02-21.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/folder-alice-2022-4-1-02-00-51.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24350/tracing/folder-brian-2022-4-1-02-00-59.zip 


</p></details>

**Review by pascalwengerter — APPROVED** [review]
E2E tests are green and code looks fine, thanks for the huge effort!

**Pull request merged**